src/main/java/ch/njol/skript/expressions/ExprTimespanDetails.java

```diff
@@ -0,0 +1,106 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Timespan Details")
+@Description("Retrieve specific information of a <a href=\"/classes.html#timespan\">timespan</a> such as hours/minutes/etc.")
+@Examples({
+	"set {_t} to difference between now and {Payouts::players::%uuid of player%::last-date}",
+	"send \"It has been %days of {_t}% day(s) since last payout.\""
+})
+@Since("INSERT VERSION")
+public class ExprTimespanDetails extends SimplePropertyExpression<Timespan, Long> {
+
+	static {
+		register(ExprTimespanDetails.class, Long.class, "(0:tick[s]|1:second[s]|2:minute[s]|3:hour[s]|4:day[s]|5:week[s]|6:month[s]|7:year[s])", "timespans");
+	}
+
+	private final int TICKS = 0, SECONDS = 1, MINUTES = 2, HOURS = 3, DAYS = 4, WEEKS = 5, MONTHS = 6, YEARS = 7;
```

Comment:
```suggestion
	enum Duration {
		TICKS(Timespan::getTicks_i),
		SECONDS(Timespan::getSeconds),
		MINUTES(Timespan::getMinutes),
		HOURS(Timespan::getHours),
		DAYS(Timespan::getDays),
		WEEKS(Timespan::getWeeks),
		MONTHS(Timespan::getMonths),
		YEARS(Timespan::getYears);
		
		private final ToIntFunction<Timespan> function;
		
		Duration(ToIntFunction<Timespan> function) {
			this.function = function;
		} 
		
		public int ticks(Timespan timespan) {
			return function.applyAsInt(timespan);
		}
	}
```
Something along these lines, the general point being you should use an enum instead of ints and then just use ordinal for the parse marks.